.github/workflows/release.yml

```diff
@@ -0,0 +1,100 @@
+name: Release
+on:
+  push:
+    tags: '*'
+
+# Needed to create the release
+permissions:
+  contents: write
+  id-token: write
+
+# Needed to create the release and use gh cli
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GH_TOKEN:  ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+    steps:
+        - name: Checkout Code
+          uses: actions/checkout@v4 
+        - name: Create a Release
+          id: create_release
+          uses: actions/create-release@v1
+          with:
+            tag_name: ${{ github.ref }}
+            release_name: Release ${{ github.ref_name }}
+            prerelease: true
+
+  build-and-push-amd64:
+    runs-on: ubuntu-latest
+    needs: [create-release]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Set up GO
+        uses: actions/setup-go@v5
```

Comment:
Not sure why we need to explicitly setup go here as it is already in our CI image. https://github.com/rancher/csp-adapter/blob/release/v2.9/Dockerfile.dapper#L1